src/event.rs

```diff
@@ -74,9 +74,94 @@ pub enum Event<'a, T: 'static> {
+    /// quits on platforms without an application lifecycle.
+    ///
+    /// In case of corner cases with how platform-specific events are mapped to Winit `Suspended`
+    /// and `Resumed` events it's recommended that applications be able to gracefully handle redundant
+    /// (i.e. back to back) `Suspended` or `Resumed` events.
+    ///
+    /// Also see [`Resumed`][Self::Resumed] notes.
+    ///
+    /// ## Android
+    ///
+    /// On Android specifically, the `Suspended` event relates to two things:
+    /// 1. The application is paused (according to the [Activity lifecycle])
+    /// 2. Or the application has had it's associated SurfaceView destroyed (which is expected
+    /// to closely correlate with being paused)
+    ///
+    /// Applications that need to run on Android should assume their SurfaceView has been
+    /// destroyed, which indirectly invalidates any existing render surfaces that may have been
+    /// created outside of Winit (such as an `EGLSurface`, [`VkSurfaceKHR`] or [`wgpu::Surface`]).
+    ///
+    /// After being `Suspended` on Android applications must drop all render surfaces before
+    /// the event callback completes, which may be re-created when the application is next `Resumed`.
+    ///
+    /// [Activity lifecycle]: https://developer.android.com/guide/components/activities/activity-lifecycle
+    /// [`VkSurfaceKHR`]: https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkSurfaceKHR.html
+    /// [`wgpu::Surface`]: https://docs.rs/wgpu/latest/wgpu/struct.Surface.html
+    ///
+    /// ## iOS
+    ///
+    /// On iOS the `Suspended` event is currently emitted in response to an
+    /// [`applicationWillResignActive`] callback which means that the application is
+    /// about to transition from the active to inactive state (according to the
+    /// [iOS application lifecycle])
+    ///
+    /// [`applicationWillResignActive`]: https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622950-applicationwillresignactive
+    /// [iOS application lifecycle]: https://developer.apple.com/documentation/uikit/app_and_environment/managing_your_app_s_life_cycle
     Suspended,
 
     /// Emitted when the application has been resumed.
+    ///
+    /// For consistency all platforms emit a `Resumed` event even if they don't themselves have a
+    /// formal suspend/resume lifecycle. For systems without a standard suspend/resume lifecycle
+    /// the `Resumed` event is is always emitted after the `NewEvents(StartCause::Init)` event.
+    ///
+    /// # Portability
+    ///
+    /// It's recommended that applications should only initialized their graphics context and create
+    /// a window after they have received their first `Resumed` event. Some systems
+    /// (specifically Android) won't allow applications to create a render surface until they are
+    /// resumed.
+    ///
+    /// In case of corner cases with how platform-specific events are mapped to Winit `Suspended`
```

Comment:
In case of corner cases?

Reply:
I ended up taking another pass at rewording this:

```rust
    /// Considering that the implementation of `Suspended` and [`Resumed`] events may be internally
    /// driven by multiple platform-specific events, and there may be subtle differences across
    /// platforms with how these internal events are delivered, it's recommended that applications
    /// be able to gracefully handle redundant (i.e. back-to-back) `Suspended` or [`Resumed`] events.
```